mage_ai/frontend/components/charts/BoxPlotHorizontal.tsx

```diff
@@ -0,0 +1,100 @@
+import ParentSize from '@visx/responsive/lib/components/ParentSize';
+import { BoxPlot } from '@visx/stats';
+import { ThemeContext } from 'styled-components';
+import { scaleLinear } from '@visx/scale';
+import { useContext } from 'react';
+
+import { ThemeType } from '@oracle/styles/themes/constants';
+
+export type BoxPlotHorizontalProps = {
+  data: {
+    min: number;
+    firstQuartile: number;
+    median: number;
+    thirdQuartile: number;
+    max: number;
+    outliers: number[];
+  },
+  primary?: boolean;
+  secondary?: boolean;
+  danger?: boolean;
+  scale?: number;
+  width?: number;
+  height?: number;
+};
+
+function BoxPlotHorizontal({
+  data,
+  primary,
+  secondary,
+  danger,
+  scale = 1.0,
+  width,
+  height,
+}: BoxPlotHorizontalProps) {
+  const themeContext: ThemeType = useContext(ThemeContext);
+
+  const colorMap = [
+    {
+      color: primary,
+      fill: themeContext.chart.backgroundPrimary,
```

Comment:
Default to a color from the `light` theme if `themeContext` is undefined. Same for below.